battenberg/utils.py

```diff
@@ -64,11 +63,38 @@ def set_initial_branch(repo: Repository, template: str):
             repo.references['HEAD'].set_target(initial_branch)
 
 
-def construct_keypair(public_key_path: str = None, private_key_path: str = None,
-                      passphrase: str = '') -> Keypair:
-    ssh_path = os.path.join(os.path.expanduser('~'), '.ssh')
-    if not public_key_path:
-        public_key_path = os.path.join(ssh_path, 'id_rsa.pub')
-    if not private_key_path:
-        private_key_path = os.path.join(ssh_path, 'id_rsa')
+ALGORITHMS = {
+    "ED25519": {
+        "public": "id_ed25519.pub",
+        "private": "id_ed25519"
+    },
+    "ED25519_SK": {
+        "public": "id_ed25519_sk.pub",
+        "private": "id_ed25519_sk"
+    },
+    "ECDSA_SK": {
+        "public": "id_ecdsa_sk.pub",
+        "private": "id_ecdsa_sk"
+    },
+    "RSA": {
+        "public": "id_rsa.pub",
+        "private": "id_rsa"
+    }
+}
+
+
+def find_key_paths(ssh_path: Path):
+    for algorithm in ALGORITHMS.values():
+        public_key_path = ssh_path / algorithm['public']
+        private_key_path = ssh_path / algorithm['private']
+        if public_key_path.exists() and private_key_path.exists():
+            return public_key_path, private_key_path
+
+    raise KeypairMissingException(f'Could not find keypair in {ssh_path}',
+                                  f'Possible options include: {ALGORITHMS}')
+
+
+def construct_keypair(ssh_path: Path = None, passphrase: str = '') -> Keypair:
+    ssh_path = ssh_path or Path('~/.ssh').expanduser()
+    public_key_path, private_key_path = find_key_paths(ssh_path)
```

Comment:
Core Issue being resolved, Gitlab allows for many Keypair algorithms, so this checks for the main types that could exists. 

`KeypairMissingException` is raised if none of the possibilities can be located.